### PR TITLE
Move FeesProvider up to the pool factory base contract

### PR DESCRIPTION
**Pull request by EndymionJkb**
All pools will now use this, so move from ComposableStable and Managed factories up into the base factory

### CODE REVIEW

File: pkg/pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol [hunk 1]
```diff
@@ -36,17 +37,20 @@ import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthenticati
  * prevent the creation of any future pools from the factory.
  */
 abstract contract BasePoolSplitCodeFactory is IBasePoolSplitCodeFactory, BaseSplitCodeFactory, SingletonAuthentication {
+    IProtocolFeePercentagesProvider private _protocolFeeProvider;
```
Comment (nventuro) [inline pkg/pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol:40]:
```suggestion
    IProtocolFeePercentagesProvider private immutable _protocolFeeProvider;
```

File: pkg/pool-linear/test/AaveLinearPoolFactory.test.ts [hunk 2]
```diff
@@ -30,7 +31,10 @@ describe('AaveLinearPoolFactory', function () {
   sharedBeforeEach('deploy factory & tokens', async () => {
     vault = await Vault.create();
     const queries = await deploy('v2-standalone-utils/BalancerQueries', { args: [vault.address] });
-    factory = await deploy('AaveLinearPoolFactory', { args: [vault.address, queries.address] });
+    protocolFeesProvider = await deploy('v2-standalone-utils/ProtocolFeePercentagesProvider', {
+      args: [vault.address, fp(1), fp(1)],
+    });
+    factory = await deploy('AaveLinearPoolFactory', { args: [vault.address, protocolFeesProvider.address, queries.address] });
```
Comment (nventuro) [inline pkg/pool-linear/test/AaveLinearPoolFactory.test.ts:37]: You can get this from the `Vault` object, no need to deploy another one

File: pkg/pool-linear/test/ERC4626LinearPoolFactory.test.ts [hunk 3]
```diff
@@ -30,7 +31,10 @@ describe('ERC4626LinearPoolFactory', function () {
 
   sharedBeforeEach('deploy factory & tokens', async () => {
     vault = await Vault.create();
-    factory = await deploy('ERC4626LinearPoolFactory', { args: [vault.address] });
+    protocolFeesProvider = await deploy('v2-standalone-utils/ProtocolFeePercentagesProvider', {
+      args: [vault.address, fp(1), fp(1)],
+    });
+    factory = await deploy('ERC4626LinearPoolFactory', { args: [vault.address, protocolFeesProvider.address] });
```
Comment (nventuro) [inline pkg/pool-linear/test/ERC4626LinearPoolFactory.test.ts:37]: Same comment re. using the Vault's

File: pkg/pool-linear/test/UnbuttonAaveLinearPoolFactory.test.ts [hunk 4]
```diff
@@ -31,7 +32,10 @@ describe('UnbuttonAaveLinearPoolFactory', function () {
 
   sharedBeforeEach('deploy factory & tokens', async () => {
     vault = await Vault.create();
-    factory = await deploy('UnbuttonAaveLinearPoolFactory', { args: [vault.address] });
+    protocolFeesProvider = await deploy('v2-standalone-utils/ProtocolFeePercentagesProvider', {
+      args: [vault.address, fp(1), fp(1)],
+    });
+    factory = await deploy('UnbuttonAaveLinearPoolFactory', { args: [vault.address, protocolFeesProvider.address] });
```
Comment (nventuro) [inline pkg/pool-linear/test/UnbuttonAaveLinearPoolFactory.test.ts:38]: And here too

File: pkg/pool-utils/test/factories/BasePoolSplitCodeFactory.test.ts [hunk 5]
```diff
@@ -25,8 +27,11 @@ describe('BasePoolSplitCodeFactory', function () {
 
     authorizer = await deploy('v2-vault/TimelockAuthorizer', { args: [admin.address, ZERO_ADDRESS, MONTH] });
     vault = await deploy('v2-vault/Vault', { args: [authorizer.address, WETH.address, MONTH, MONTH] });
+    protocolFeesProvider = await deploy('v2-standalone-utils/ProtocolFeePercentagesProvider', {
+      args: [vault.address, fp(1), fp(1)],
+    });
 
-    factory = await deploy('MockPoolSplitCodeFactory', { args: [vault.address] });
+    factory = await deploy('MockPoolSplitCodeFactory', { args: [vault.address, protocolFeesProvider.address] });
```
Comment (nventuro) [inline pkg/pool-utils/test/factories/BasePoolSplitCodeFactory.test.ts:34]: Once more :)

File: pkg/pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/pool-utils/contracts/factories/BasePoolSplitCodeFactory.sol:40]: I wondered why it wasn't immutable in the other contracts

File: pkg/pool-utils/test/factories/BasePoolSplitCodeFactory.test.ts (hunk 5 above)
Comment (EndymionJkb) [reply pkg/pool-utils/test/factories/BasePoolSplitCodeFactory.test.ts:34]: Vault.ts deploys and sets it, but if you deploy the Vault directly, you get the Vault contract itself. I did find another case in benchmarks, though

**Review by nventuro — APPROVED** [review]
Fantastic!

**Pull request merged**